polaris-core/src/main/java/org/apache/polaris/core/config/PolarisConfigurationStore.java

```diff
@@ -99,7 +99,25 @@ public interface PolarisConfigurationStore {
    */
   default <T> @Nonnull T getConfiguration(
       @Nonnull RealmContext realmContext, PolarisConfiguration<T> config) {
-    T result = getConfiguration(realmContext, config.key, config.defaultValue);
+    T result = getConfiguration(realmContext, config.key);
+
+    if (result == null && config.hasLegacyKeys()) {
+      for (String legacyKey : config.legacyKeys) {
```

Comment:
If `legacyKeys` just returned a set (or there was a `keys` that returned a set) then the if statement here goes away.

Reply:
Done.